rasterio/rio/pyramid.py

```diff
@@ -0,0 +1,75 @@
+"""Manage overviews of a dataset."""
+
+import logging
+
+import click
+
+import rasterio
+from rasterio.enums import Resampling
+
+from . import options
+
+
+def build_handler(ctx, param, value):
+    if value:
+        try:
+            if '^' in value:
+                base, exp_range = value.split('^')
+                exp_min, exp_max = (int(v) for v in exp_range.split('..'))
+                value = [pow(int(base), k) for k in range(exp_min, exp_max+1)]
+            else:
+                value = [int(v) for v in value.split(',')]
+        except Exception as exc:
+            raise click.BadParameter("must match 'n,n,n,...' or 'n^n..n'.")
+    return value
+
+@click.command('pyramid', short_help="Construct overviews in an existing dataset.")
+@options.file_in_arg
+@click.option('--build', callback=build_handler,
```

Comment:
@sgillies What do you think about setting `metavar="n,n,...|n^n..n"` here?  Otherwise its only listed in the longer help text and I find its nice to have non-standard syntaxes in an easy to find location.